lms/djangoapps/grades/new/course_grade.py

```diff
@@ -24,8 +26,36 @@ class CourseGrade(object):
     def __init__(self, student, course, course_structure):
         self.student = student
         self.course = course
+        self.course_version = getattr(course, 'course_version', None)
+        self.course_edited_timestamp = getattr(course, 'subtree_edited_on', None)
         self.course_structure = course_structure
         self.chapter_grades = []
+        self._percent = None
+        self._letter_grade = None
+
+    @classmethod
+    def init_from_model(cls, user, course, course_structure, current_grading_policy_hash):
+        """
+        Initializes a CourseGrade object, filling its members with persisted values from the database.
+
+        If no persisted values are found, returns None.
+        """
+        try:
+            persistent_grade = PersistentCourseGrade.read_course_grade(user.id, course.id)
+        except PersistentCourseGrade.DoesNotExist:
+            return None
+        course_grade = CourseGrade(user, course, course_structure)
+
+        if current_grading_policy_hash != persistent_grade.grading_policy_hash:
+            course_grade.compute_and_update(read_only=False)
```

Comment:
Is this going to have the same performance issue we have with subsection grades?